### PR TITLE
Bump open api generator version

### PR DESCRIPTION
**Pull request by mirkokurt**
### Motivation

The version of the open API generator used is very old. We want to use a recent version (7.0.0)

### Change description

- Open API generator bumped to 7.0.0 for every language and for the docs
- Template-patches updated to be applied to the updated template files

### Reviewer checklist

* [ ] PR address a single concern.
* [ ] PR title and description are properly filled.
* [ ] Changes will be merged in `master`.
* [ ] Changes are covered by tests.
* [ ] Logging is meaningful in case of troubleshooting.
* [ ] History is clean, commit messages are meaningful (see `CONTRIBUTING.md`) and are well formatted.
* [ ] Mentioned the Analytics team when there are schema changes.

### CODE REVIEW

File: template-patches/python-configuration.patch [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-index dceab63..b2d56ab 100644
+index ac51800..2511102 100644
 --- a/python/configuration.mustache
 +++ b/python/configuration.mustache
-@@ -259,7 +259,7 @@ conf = {{{packageName}}}.Configuration(
+@@ -265,7 +265,7 @@ conf = {{{packageName}}}.Configuration(
          """Adding retries to override urllib3 default value 3
          """
-         # Disable client side validation
+         # Enable client side validation
```
Comment (robgee86) [inline template-patches/python-configuration.patch:8]:
```suggestion
         # Disable client side validation
```

Comment (mirkokurt) [reply template-patches/python-configuration.patch:8]: 👍

File: template-patches/python-configuration.patch [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 -        self.client_side_validation = True
 +        self.client_side_validation = False
  
-     def __deepcopy__(self, memo):
-         cls = self.__class__
+         self.socket_options = None
+         """Options to pass down to the underlying urllib3 socket
```
Comment (robgee86) [inline template-patches/python-configuration.patch:13]: This is going to replace lines 12 and 13 in the old template, can they really be removed?

Also, I remember that clients generated with the new version of openapi-generator already had client side validation disabled so maybe this template is not necessary anymore?

Comment (mirkokurt) [reply template-patches/python-configuration.patch:13]: No, the patch isn't replacing the lines you highlighted. Maybe the combination of GitHub diffs and patch diffs is preventing you from reading clearly, if you pull the source code and check you'll see that those lines aren't removed.

Comment (robgee86) [reply template-patches/python-configuration.patch:13]: Yes, right, I forgot this is a diff.

As for the second point: do we still need to apply this patch?

Comment (mirkokurt) [reply template-patches/python-configuration.patch:13]: Yes, unfortunately client-side validation it's still active in the 7.0.0 https://github.com/OpenAPITools/openapi-generator/blob/c37fa8a4e8399ebd76d7f190136fd4b851c9aa7e/modules/openapi-generator/src/main/resources/python/configuration.mustache#L268

File: template-patches/python-configuration.patch (hunk 1 above)
Comment (mirkokurt) [reply template-patches/python-configuration.patch]: done here: 9a26213693a7c304c53518b5abc74d79ed13a286

**Review by robgee86 — APPROVED** [review]
Awesome!

**Pull request merged**